### PR TITLE
fix: magazines dropped by zombies vary in ammocount, expand variety of guns

### PR DESCRIPTION
**Pull request by chaosvolt**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

CrystalNole over on the BN discord discovered that basic zombies were always dropping fully-loaded magazines when they drop ammo, when the guns an obviously-failed survivor spawned with logically should have some chance of being depleted. Along the way, I decided to also tweak it so the variety of guns that can spawn has a bit wider variety.

## Describe the solution

1. Removed the default 100 ammo rate for carried guns, needed to allow the following to work.
2. Added `charges-min` of zero to all mag spawns in carried itemgroups. Found this works fine to make magazine spawns, even for magazine groups, vary in ammo spawned even when the same group is used by a stored gun spawn to spawn a full mag.
3. As a bonus, reworked the guns that basic zeds can spawn with. Instead of only ever spawning with pistol or shotgun guns and ammo, I created a custom itemgroup that picks between pistols, shotguns, and rifles (with pistols being picked 50% of the time). The guns also have a small chance of using the improvised counterpart instead. Same vein as NPCs defaulting to sometimes picking improvised guns, with desperate rioters arming themselves with anything they could in the months prior to the cataclysm as well as potentially some post-cata survivors who eventually succumbed. Likewise for the ammo groups, it has a small chance of picking the reloaded itemgroup instead, meaning blackpowder loads can spawn on occasion when it picks those.
4. Misc: Fixed one of the carried gun groups spawning the gun with invalid ammo.

## Describe alternatives you've considered

screm

## Testing

1. Checked affected files for syntax and lint errors.
2. Load-tested in compiled test build.
3. Spawned in a large number of basic zombies and debug-killed them.
4. Sorted their guns, ammo, and mag spawns off into sorting zones and examined contents, results look as expected.
5. Repeated with some soldier zombies to make sure their spawns didn't get fucked up.

<details>
<summary>Screenshots:</summary>

Ammo dropped by the group of zombies:
image
image

Guns dropped by the group of zombies:
image
image

Mags dropped by the group of zombies:
image
image

</details>

### CODE REVIEW

**Comment by Lamandus** [comment]
Are uncommon or rare guns also included in the spawns? Or just the common group.
We have a lot of guns that are super rare.

**Comment by chaosvolt** [comment]
Currently only common and improvised.

**Review by scarf005 — APPROVED** [review]

File: data/json/monsterdrops/zombie_default.json [hunk 1]
```diff
@@ -229,14 +229,30 @@
...
       [ "postman_gear", 5 ],
       [ "hardware_plumbing", 5 ]
     ]
   },
+  {
+    "type": "item_group",
+    "subtype": "distribution",
+    "id": "default_zombie_carried_guns",
+    "entries": [
+      { "group": "carried_guns_pistol_common", "prob": 45 },
```
Comment (Lamandus) [inline data/json/monsterdrops/zombie_default.json:242]:
```suggestion
      { "group": "carried_guns_pistol_common", "prob": 45 },
      { "group": "carried_guns_pistol_rare", "prob": 4 },
      { "group": "carried_guns_pistol_obscure", "prob": 2 },
```

File: data/json/monsterdrops/zombie_default.json [hunk 2]
```diff
@@ -229,14 +229,30 @@
...
+      { "group": "carried_guns_shotgun_common", "prob": 20 },
+      { "group": "carried_guns_rifle_common", "prob": 20 },
+      { "group": "guns_pistol_improvised", "prob": 5 },
+      { "group": "guns_shotgun_improvised", "prob": 5 },
+      { "group": "guns_rifle_improvised", "prob": 5 },
+      { "group": "ammo_pistol_common", "prob": 45 },
```
Comment (Lamandus) [inline data/json/monsterdrops/zombie_default.json:248]:
```suggestion
      { "group": "ammo_pistol_common", "prob": 45 },
      { "group": "ammo_pistol_rare", "prob": 4 },
      { "group": "ammo_pistol_obscure", "prob": 2 },
```

Comment (Lamandus) [reply data/json/monsterdrops/zombie_default.json:248]: for adding more pistol variations.

Shotguns and rifles in rare and obscure are more milispec than regular (SPAS, Famas, etc.)

Comment (Lamandus) [reply data/json/monsterdrops/zombie_default.json:248]: also smgs should spawn here and there.

File: data/json/monsterdrops/zombie_default.json [hunk 3]
```diff
@@ -229,14 +229,30 @@
...
   },
+  {
+    "type": "item_group",
+    "subtype": "distribution",
+    "id": "default_zombie_carried_guns",
+    "entries": [
+      { "group": "carried_guns_pistol_common", "prob": 45 },
+      { "group": "carried_guns_shotgun_common", "prob": 20 },
+      { "group": "carried_guns_rifle_common", "prob": 20 },
+      { "group": "guns_pistol_improvised", "prob": 5 },
```
Comment (Lamandus) [inline data/json/monsterdrops/zombie_default.json:245]:
```suggestion
      { "group": "guns_pistol_improvised", "prob": 5 },
      { "group": "guns_smg_improvised", "prob": 5 },
```

**Comment by chaosvolt** [comment]
Actually, let's merge fixes now and expand the gun variety in a follow-up.

**Pull request merged**